lxd/identities.go

```diff
@@ -1061,26 +1040,43 @@ func deleteIdentity(d *Daemon, r *http.Request) response.Response {
 	}
 
 	// Notify other cluster members to update their identity cache.
-	notifier, err := cluster.NewNotifier(s, s.Endpoints.NetworkCert(), s.ServerCert(), cluster.NotifyAlive)
+	notify := newIdentityNotificationFunc(s, r, s.Endpoints.NetworkCert(), s.ServerCert())
+	_, err = notify(lifecycle.IdentityDeleted, string(id.AuthMethod), id.Identifier, true)
 	if err != nil {
 		return response.SmartError(err)
 	}
 
-	err = notifier(func(client lxd.InstanceServer) error {
-		_, _, err := client.RawQuery(http.MethodPost, "/internal/identity-cache-refresh", nil, "")
-		return err
-	})
-	if err != nil {
-		return response.SmartError(err)
-	}
+	return response.EmptySyncResponse
+}
 
-	// Send a lifecycle event for the identity update.
-	lc := lifecycle.IdentityDeleted.Event(string(id.AuthMethod), id.Identifier, request.CreateRequestor(r), nil)
-	s.Events.SendLifecycle(api.ProjectDefaultName, lc)
+// newIdentityNotificationFunc returns a function that creates and sends a lifecycle event for the identity.
+// If updateCache is true, the local identity cache is updated and a notification is sent to other members to do the same.
+func newIdentityNotificationFunc(s *state.State, r *http.Request, networkCert *shared.CertInfo, serverCert *shared.CertInfo) identityNotificationFunc {
+	return func(action lifecycle.IdentityAction, authenticationMethod string, identifier string, updateCache bool) (*api.EventLifecycle, error) {
```

Comment:
when wouldnt we update the cache, why is it optional?

Reply:
When a pending identity is created we emit a lifecycle event because it's something that is accessible and manageable via the API, but we don't need to update the identity cache.

Reply:
understood thanks